### PR TITLE
add auto goimports into Makefile

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: Yang Keao <keao.yang@yahoo.com>

### What problem does this PR solve?

Add `goimports` command into Makefile.

### What is changed and how does it work?

Diff check has been added to CI to ensure every PR has been formatted and yaml has been regenerated.

Tests 

 - No code

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -86,9 +86,12 @@ manifests: controller-gen
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 
 # Run go fmt against code
-fmt:
+fmt: groupimports
 	$(GO) fmt ./...
 
+groupimports:
```
Comment (zhouqiang-cl) [inline Makefile]: Did we need to install goimport first

**Comment by Yisaer** [comment]
We should also add this into the CI test.

File: Makefile (hunk 1 above)
Comment (YangKeao) [reply Makefile]: Yes! Though `goimports` is auto installed as `golang.org/x/tools` is an indirect dependency, it's still necessary to check and install.

**Pull request merged**